recipe-client-addon/lib/PreferenceRollout.jsm

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+"use strict";
+
+const {utils: Cu} = Components;
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "Services", "resource://gre/modules/Services.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "CleanupManager", "resource://shield-recipe-client/lib/CleanupManager.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "JSONFile", "resource://gre/modules/JSONFile.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "OS", "resource://gre/modules/osfile.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "LogManager", "resource://shield-recipe-client/lib/LogManager.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Preferences", "resource://gre/modules/Preferences.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "TelemetryEnvironment", "resource://gre/modules/TelemetryEnvironment.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "PreferenceManagement", "resource://shield-recipe-client/lib/PreferenceManagement.jsm");
+
+this.EXPORTED_SYMBOLS = ["PreferenceRollout"];
+
+const log = LogManager.getLogger("preference-rollouts");
+
+const PREF_NAMESPACE = "rollouts";
+
+this.PreferenceRollout = Object.assign({}, PreferenceManagement(PREF_NAMESPACE), {
+  async init() {
+    this.applyAll();
+  },
+
+  async start({name, preferenceName, preferenceValue, preferenceBranch, preferenceType}) {
+    log.debug(`PreferenceRollout.enroll(${name} - ${preferenceBranch}/${preferenceName})`);
+
+    const hasExistingRollout = await this.has(name);
+    if (hasExistingRollout) {
+      throw new Error(`A preference rollout named "${name}" already exists.`);
+    }
+
+    /** @type {Experiment} */
+    const experiment = {
+      name,
+      preferenceName,
+      preferenceValue,
+      preferenceType,
+      preferenceBranch,
+      branch: preferenceBranch,
```

Comment:
Can we make this API use the same parameter names as the PreferenceExperiments one?